### PR TITLE
support for multidisk OVA files

### PR DESCRIPTION
**Pull request by abhinandanprateek**
Signed-off-by: Abhinandan Prateek <abhinandan.prateek@shapeblue.com>

### CODE REVIEW

**Comment by borisstoyanov** [comment]
@abhinandanprateek shouldn't this be in the Interoute repo? Can you please move the PR there?

**Comment by abhinandanprateek** [comment]
@borisstoyanov AFAIK other than the two FR that @rhtyd  is working on, others need to go to master cc @PaulAngus

**Comment by borisstoyanov** [comment]
I see @DaanHoogland has created the PR for FR2.6 in the Interoute repo as well, is there any concerns with submitting the PRs from the interoute repo? 

Actually what's the strategy on the community submissions? Where do we create PRs and how do we submit them in community? 
cc: @rhtyd @PaulAngus @DaanHoogland

**Comment by yadvr** [comment]
@abhinandanprateek @borisstoyanov /cc @PaulAngus 

I've implemented the following UI, where in case of multi-disk template (ova) you'll see options to select the disk offering for the data disk template:

screenshot from 2017-04-12 17-46-52

Though this will take away the option to deploy additional data disk with the template. The UI code sort of existed already in the wizard, I found this today and made some changes to the API and the UI to reuse the widget which is a list of checkbox with labels (the data disk name) and the list of disk offerings per data disk.

The params will be passed to the API as follows:
screenshot from 2017-04-12 18-00-13

**Comment by yadvr** [comment]
Additional note: this may need some refining as it does not make sense to me to select a disk offering that has a fixed size where the data disk (template) might already have a fixed size?

**Comment by borisstoyanov** [comment]
Hi guys, looks like there are conflicts in the data schema, can we resolve them before jumping into testing please? @abhinandanprateek @rhtyd

**Comment by borisstoyanov** [comment]
Hi guys tried building the PR but build failed with 
```[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 11:43.636s
[INFO] Finished at: Wed Apr 19 06:17:41 UTC 2017
[INFO] Final Memory: 113M/1448M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.2:compile (default-compile) on project cloud-vmware-base: Compilation failure: Compilation failure:
[ERROR] /jenkins/workspace/shapeblue-centos6-pkg-builder/dist/rpmbuild/BUILD/cloudstack-4.10.0.0-SNAPSHOT/vmware-base/src/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java:[164,15] error: cannot find symbol
[ERROR] symbol:   variable UUID
[ERROR] location: class HypervisorHostHelper
[ERROR] /jenkins/workspace/shapeblue-centos6-pkg-builder/dist/rpmbuild/BUILD/cloudstack-4.10.0.0-SNAPSHOT/vmware-base/src/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java:[1662,34] error: cannot find symbol
[ERROR] symbol:   variable UUID
[ERROR] location: class HypervisorHostHelper
[ERROR] /jenkins/workspace/shapeblue-centos6-pkg-builder/dist/rpmbuild/BUILD/cloudstack-4.10.0.0-SNAPSHOT/vmware-base/src/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java:[1772,16] error: cannot find symbol
[ERROR] symbol:   class FileWriter
[ERROR] location: class HypervisorHostHelper
[ERROR] /jenkins/workspace/shapeblue-centos6-pkg-builder/dist/rpmbuild/BUILD/cloudstack-4.10.0.0-SNAPSHOT/vmware-base/src/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java:[1772,37] error: cannot find symbol
[ERROR] -> [Help 1]
[ERROR] 
```
http://build.yadav.xyz:8080/job/shapeblue-pkg-builder/60/
Is this related to the removal of midonet plugin?
@abhinandanprateek @rhtyd

**Comment by yadvr** [comment]
Yes @borisstoyanov I've sent a PR to fix the master/upstream issue

File: vmware-base/src/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java [hunk 1]
```diff
@@ -1637,6 +1653,203 @@ public void action(Long param) {
...
+        assert (morRp != null);
+        ManagedObjectReference morHost = host.getMor();
+        String importEntityName = UUID.randomUUID().toString();
+        OvfCreateImportSpecParams importSpecParams = new OvfCreateImportSpecParams();
+        importSpecParams.setHostSystem(morHost);
+        importSpecParams.setLocale("US");
+        importSpecParams.setEntityName(importEntityName);
+        importSpecParams.setDeploymentOption("");
+
+        String ovfDescriptor = removeOVFNetwork(HttpNfcLeaseMO.readOvfContent(ovfFilePath));
```
Comment (yadvr) [inline vmware-base/src/com/cloud/hypervisor/vmware/mo/HypervisorHostHelper.java:1673]: @abhinandanprateek I've made this additional change

**Comment by borisstoyanov** [comment]
packages here: 
http://staging.yadav.xyz/cloudstack/custom/Int-FR2.3/

**Comment by abhinandanprateek** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@abhinandanprateek Did you merge back master? I think you better rebase, instead. This is going to look messy.

**Comment by abhinandanprateek** [comment]
@DaanHoogland heard ya

**Comment by abhinandanprateek** [comment]
moved to community.

**Pull request closed without merging**